specs/checkbox.md

```diff
@@ -0,0 +1,128 @@
+# FAST-Checkbox
+
+## Overview
+An implementation of a checkbox as a form-connected web-component.
+
+### Use Cases
+Used anywhere an author might otherwise use an `input[type="checkbox"]`
+  
+### Features
+- form association
+- focus delegation
+
+### Risks and Challenges
+We want general feature-parity between this component and an `input[type="checkbox"]`. Most of these concerns are addressed by https://github.com/microsoft/fast-dna/pull/2723, but some stateful nuance will need to be handled by this component.
+
+### Prior Art/Examples
+- [FAST-DNA (React)](https://explore.fast.design/components/checkbox)
+- [Material UI](https://material-ui.com/components/checkboxes/)
+- [Lightning Design](https://www.lightningdesignsystem.com/components/checkbox/)
+- [Carbon Design](https://www.carbondesignsystem.com/components/checkbox/code)
+- [Ant Design](https://ant.design/components/checkbox/)
+- [Atlassian](https://atlaskit.atlassian.com/packages/core/checkbox)
+- [Office Fabric](https://developer.microsoft.com/en-us/fabric#/controls/web/checkbox)
+- [Windows (UWP)](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/checkbox)
+---
+
+### API
+Extends [form associated custom element](./form-associated-custom-element.md).
+
+*Component Name*
+`fast-checkbox`
+
+*IDL attributes*
+- `defaultChecked: boolean`
+  - If the checkbox is checked by default. Synchronized with the `checked` content attribute
+- `checked: boolean`
+  - The current checked state of the checkbox
+- `indeterminate: boolean`
+  - The indeterminate state. Independent of checked
+
+*Content attributes*
+- `readonly`
+  - The checkbox should be submitted with the form but should not be editable.
+- `value`
+  - Defaults to "on" to match `input[type="checkbox"]`
+- `checked`
+  - The initial checked value. 
+  
+*Events*
+- `change: CustomEvent`
+  - no custom data
+  - bubbles
+
+### Anatomy and Appearance
+
+```HTML
+<!-- shadow root -->
+<label part="label"><slot></slot></label>
+<div part="root">
```

Comment:
It's interesting that the "root" in this case is not the wrapper that includes the label. Is our convention going forward that root does not necessarily include other pieces of UI? I would assume that switch would have a root so I could refer to another component with a similar concept of labelling but there's no root on there (I think maybe that was a miss)

Reply:
Yea this was meant to spark conversation :)

the "root" of this component is the custom element, `:host` - in this instance we don't really need a wrapping element for the whole component, but the going convention seems to be to have a "root" part. Really, this is just the "checkbox" portion, but `fast-checkbox::part(checkbox)` also feels strange.

Reply:
interesting - if we don't need a wrapping element, I'm all in favor of this approach. I'm going to update my two PR's with this. One question this does bring up though is "how do we decide what is the 'root'"?

Reply:
I think since this is primarily for styling there should be a visual block aspect to it, similar to what's going on here. When we refer to the checkbox we refer to the control that looks like a box with a check in it, not necessarily any extra text or popups etc that may come along with it. That said, if there comes a need to refactor and add a wrapper to the component and we've already used the word "root", then what should that wrapper be called? Perhaps "root" should refer to the wrapping instance we should use the word "control" should refer to the actual interactive piece.

Reply:
See changes - this changed a bit while working through the implementation.